packages/commonwealth/client/scripts/state/api/comments/deleteReaction.ts

```diff
@@ -0,0 +1,85 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import axios from 'axios';
+import ReactionCount from 'models/ReactionCount';
+import app from 'state';
+import { ApiEndpoints } from 'state/api/config';
+import useFetchCommentReactionsQuery from './fetchReaction';
+
+interface DeleteReactionProps {
+  reactionCount: ReactionCount<any>
+  canvasHash: string
+  reactionId: number;
+}
+
+const deleteReaction = async ({
+  reactionCount,
+  canvasHash,
+  reactionId
+}: DeleteReactionProps) => {
+  const {
+    session = null,
+    action = null,
+    hash = null,
+  } = await app.sessions.signDeleteCommentReaction({
+    comment_id: canvasHash,
+  })
+
+  return await axios.delete(`${app.serverUrl()}/reactions/${reactionId}`, {
+    data: {
+      jwt: app.user.jwt,
+      canvas_action: action,
+      canvas_session: session,
+      canvas_hash: hash,
+    },
+  }).then((r) => ({
+    ...r,
+    data: {
+      ...r.data,
+      result: {
+        ...(r.data.result || {}),
+        reactionCount,
+        reactionId
+      }
+    }
+  }));
+};
+
+interface IuseDeleteCommentReactionMutation {
+  chainId: string;
+  commentId: number;
+}
+
+const useDeleteCommentReactionMutation = ({ commentId, chainId }: IuseDeleteCommentReactionMutation) => {
+  const queryClient = useQueryClient();
+  const { data: reactions } = useFetchCommentReactionsQuery({
+    chainId,
+    commentId: commentId as number,
+  })
+
+  return useMutation({
+    mutationFn: deleteReaction,
+    onSuccess: async (response) => {
+      const { reactionCount, reactionId } = response.data.result
+
+      // update fetch reaction query state
+      const key = [ApiEndpoints.getCommentReactions(commentId), chainId]
+      queryClient.cancelQueries({ queryKey: key });
+      queryClient.setQueryData([...key],
+        () => {
+          const updatedReactions = [...(reactions || []).filter(x => x.id !== reactionId)]
+          return updatedReactions
+        }
+      );
```

Comment:
is this an optimistic update part, right?

Reply:
No, we are actually updating the exiting comments reaction already fetched by the `/comments/:id/reactions` API (which we now have in react query).

There are 2 options here, 
A- either invalidate the existing cache completely which would trigger an API call to fetch the fresh data
B- update cache with the new data (this is best if we know what the final data looks like)

Since we already have the data, so instead of Option A i.e `queryClient.invalidateQueries` (which would remove the cache and trigger the API call for the invalidated key), I just updated the existing cache (Option B). 

We can invalidate the cache, but I don't recommend that.

Reply:
thanks for explaining this. Yes, if we have data available we can manipulate the cache and avoid another server trip